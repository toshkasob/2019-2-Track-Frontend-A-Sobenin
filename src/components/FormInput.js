/* eslint-disable react/no-unused-prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import styles from '../styles/formInputStyles.module.scss';

// import { ReactComponent as AttachSvg } from '../images/forminput_buttons/attach_button.svg';
import { ReactComponent as SubmitSvg } from '../images/button_submit.svg';

export default function FormInput(props) {
	const curProps = props;

	return (
		<div className={styles.form_input}>
			<input
				id="input"
				className={styles.input}
				type="text"
				value={curProps.value}
				placeholder={curProps.placeholder}
				onChange={curProps.onChange}
			/>
			<button
				type="submit"
				className={styles.button_submit}
				style={{ display: curProps.submitButtonDisplayStyle }}
			>
				<SubmitSvg className={styles.button_submit_img} />
			</button>
		</div>
	);
}

FormInput.propTypes = {
	submitButtonDisplayStyle: PropTypes.string.isRequired,
	onChange: PropTypes.func.isRequired,
	placeholder: PropTypes.string.isRequired,
	value: PropTypes.string.isRequired,
};
