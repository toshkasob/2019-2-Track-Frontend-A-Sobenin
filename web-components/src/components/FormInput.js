const template = document.createElement('template')
template.innerHTML = `
    <style>
        input {
            border: 0;
            outline: none;
            width: calc(100% - 2px);
            height: 100%;
            font-size: 25px;
            align-item: center;
            border-top: 3px solid rgba(25, 25, 25, 0.32);
            padding-left: 5vw;
            padding-right: 5vw;
            box-sizing: border-box;
        }

        :host {
            display: inline-block;
            border: 1px solid rgba(25, 25, 25, 0.32);
        }
    </style>
    <input type="text">
`

class FormInput extends HTMLElement {
  constructor() {
    super()
    this.shadowRoot = this.attachShadow({ mode: 'open' })
    this.shadowRoot.appendChild(template.content.cloneNode(true))

    this.$input = this.shadowRoot.querySelector('input')
  }

  static get observedAttributes() {
    return ['name', 'value', 'placeholder', 'disabled']
  }

  attributeChangedCallback(name, oldValue, newValue) {
    this.$input.setAttribute(name, newValue)
  }

  set value(_newValue) {
    this.$input.value = _newValue
  }

  get value() {
    return this.$input.value
  }
}

customElements.define('form-input', FormInput)
