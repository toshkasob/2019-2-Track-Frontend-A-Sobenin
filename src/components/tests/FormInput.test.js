import React from 'react';
import renderer from 'react-test-renderer';

import FormInput from '../FormInput';

const fnMock = jest.fn();
it('renders FormInput without audio record correctly', () => {
	const frmInp = renderer
		.create(
			<div>
				<FormInput
					placeholder="Сообщение"
					value="No audio record"
					onChange={fnMock}
					attachGeo={fnMock}
					handleImage={fnMock}
					handleRecordAudio={fnMock}
					handleStopRecord={fnMock}
					flagRec="false"
					submitButtonDisplayStyle="inline-block"
				/>
			</div>,
		)
		.toJSON();
	expect(frmInp).toMatchSnapshot();
});
