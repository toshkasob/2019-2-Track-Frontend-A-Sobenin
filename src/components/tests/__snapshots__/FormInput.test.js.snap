// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FormInput without audio record correctly 1`] = `
<div>
  <div
    className="form_input"
  >
    <input
      className="input"
      id="input"
      onChange={[MockFunction]}
      placeholder="Сообщение"
      type="text"
      value="No audio record"
    />
    <button
      className="button_attach"
      id="get-geo"
      onClick={[MockFunction]}
      type="button"
    >
      <svg
        className="button_attach_img"
      >
        button_geo.svg
      </svg>
    </button>
    <button
      className="button_attach"
      id="attach-image"
      onClick={[Function]}
      type="button"
    >
      <svg
        className="button_attach_img"
      >
        button_attach_image.svg
      </svg>
    </button>
    <input
      accept="image/*"
      onChange={[MockFunction]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <button
      className="button_attach"
      id="start"
      onClick={[MockFunction]}
      style={
        Object {
          "display": undefined,
        }
      }
      type="button"
    >
      <svg
        className="button_attach_img"
      >
        button_rec_audio.svg
      </svg>
    </button>
    <button
      className="button_attach"
      id="stop"
      onClick={[MockFunction]}
      style={
        Object {
          "display": undefined,
        }
      }
      type="button"
    >
      <svg
        className="button_attach_img"
      >
        button_stop_audio.svg
      </svg>
    </button>
    <button
      className="button_submit"
      style={
        Object {
          "display": "inline-block",
        }
      }
      type="submit"
    >
      <svg
        className="button_submit_img"
      >
        button_submit.svg
      </svg>
    </button>
  </div>
</div>
`;
